iothub/service/src/Amqp/AmqpClientHelper.cs

```diff
@@ -130,41 +127,41 @@ public static Exception ToIotHubClientContract(Error error)
             {
-                retException = new MessageTooLargeException(message);
+                retException = new IotHubServiceException(HttpStatusCode.RequestEntityTooLarge, IotHubErrorCode.MessageTooLarge, message);
             }
             else if (error.Condition.Equals(AmqpErrorCode.ResourceLimitExceeded))
             {
-                retException = new DeviceMaximumQueueDepthExceededException(message);
+                retException = new IotHubServiceException(HttpStatusCode.Forbidden, IotHubErrorCode.DeviceMaximumQueueDepthExceeded, message);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.DeviceAlreadyExists))
             {
-                retException = new DeviceAlreadyExistsException(message, (Exception)null);
+                retException = new IotHubServiceException(HttpStatusCode.Conflict, IotHubErrorCode.DeviceAlreadyExists, message, (Exception)null);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.DeviceContainerThrottled))
             {
-                retException = new IotHubThrottledException(message, (Exception)null);
+                retException = new IotHubServiceException((HttpStatusCode)429, IotHubErrorCode.ThrottlingException, message, (Exception)null);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.QuotaExceeded))
             {
-                retException = new QuotaExceededException(message, (Exception)null);
+                retException = new IotHubServiceException(HttpStatusCode.Forbidden, IotHubErrorCode.IotHubQuotaExceeded, message, (Exception)null);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.PreconditionFailed))
             {
-                retException = new PreconditionFailedException(message, (Exception)null);
+                retException = new IotHubServiceException(HttpStatusCode.PreconditionFailed, IotHubErrorCode.PreconditionFailed, message, (Exception)null);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.IotHubSuspended))
             {
-                retException = new IotHubSuspendedException(message);
+                retException = new IotHubServiceException(HttpStatusCode.BadRequest, IotHubErrorCode.IotHubSuspended, message);
             }
             else
             {
-                retException = new IotHubException(message);
+                retException = new IotHubServiceException(message);
             }
 
             if (trackingId != null
-                && retException is IotHubException exHub)
+                && retException is IotHubServiceException exHub)
             {
-                IotHubException iotHubException = exHub;
+                IotHubServiceException iotHubException = exHub;
                 iotHubException.TrackingId = trackingId;
                 // This is created but not assigned to `retException`. If we change that now, it might be a
```

Comment:
Here's the breaking change note I mentioned in Teams

These code is executing but does nothing. It should be put on the exception that is returned